### PR TITLE
🍞 

### LINKED ISSUE #4: Give user some kind of alert that data has been copied

### LINKED ISSUE #31: Should show user that a PDF is being fetched, and something else when it's done
Even if no results are added to the comment, we need a way to show the user that we're done looking

### PR DESCRIPTION
**Pull request by adirar111**
resolves #4 
related to #31 

gg

### CODE REVIEW

File: extension/ts/toast.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+		@-webkit-keyframes fadeout {
+			from {top: 80px; opacity: 1;}
+			to {top: 0; opacity: 0;}
+		}
+
+		@keyframes fadeout {
+			from {top: 80px; opacity: 1;}
+			to {top: 0; opacity: 0;}
+		}
```
Comment (adirar111) [inline extension/ts/toast.ts:55]: stolen from some w3 page

File: extension/ts/toast.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	const head = document.head || document.getElementsByTagName("head")[0];
+	const style = document.createElement("style");
+	style.appendChild(document.createTextNode(cssText));
+	head.appendChild(style);
+	return style;
+}
+
+const createToast = (toastType: ToastType) => {
+	const toast = document.createElement("div");
+	toast.id = `better-library-thing-toast-${Math.floor(Math.random() * 100)}`;
```
Comment (adirar111) [inline extension/ts/toast.ts:16]: to handle multiple toasts at the same time (conflicts with the generated number, if any, should be rare)

File: extension/ts/toast.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	return {style, toast};
+};
+
+export const showToast = (text: string, toastType: ToastType) => {
+	const {style, toast} = createToast(toastType);
+	toast.innerText = text;
+	setTimeout(() => {
+		const head = document.head || document.getElementsByTagName("head")[0];
+		document.body.removeChild(toast);
+		head.removeChild(style);
```
Comment (adirar111) [inline extension/ts/toast.ts:67]: clean up

**Review by braxtonhall — COMMENTED** [review]
VERY hype!

I think it would also make sense to wrap the load feature in a try/catch and display a toast if there's an error, JUUUST in case our localStorage got mutated somehow?

**Comment by braxtonhall** [comment]
ALSO, I would mark this as being related to #31, but not explicitly resolving it, as a progress indicator is important too.

**Comment by braxtonhall** [comment]
ALSO??? how tf did u make that gif????

File: extension/ts/toast.ts (hunk 2 above)
Comment (braxtonhall) [reply extension/ts/toast.ts:16]: Actually, couldn't you just increment a counter? The counter "resets" when you navigate away from the page or reset.

**Comment by adirar111** [comment]
> ALSO??? how tf did u make that gif????

kap is the :goat:, no 🧢  -> https://getkap.co/

**Review by adirar111 — COMMENTED** [review]
this is a tad overengineered

**Review by braxtonhall — APPROVED** [review]
I think this is good to merge. I may PR some cosmetic changes on top after you merge tho -- will detail thoughts on all of them to see what you think

**Pull request merged**